File: samples/java/src/main/java/com/example/account/BankAccount.java
// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>

package com.example.account;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.util.UUID;

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record BankAccount(UUID accountNumber, String accountOwner, String securityCode, double balance) {
}

File: samples/scala/src/main/scala/com/example/account/BankAccount.scala
package com.example.account

import play.api.libs.json.{Format, Json}

import java.util.UUID

object BankAccount {
  implicit val format: Format[BankAccount] = Json.format
}

case class BankAccount(accountNumber: UUID, accountOwner: String, securityCode: String, balance: Double)